beacon-chain/blockchain/chain_info.go

```diff
@@ -273,13 +273,11 @@ func (s *Service) CurrentFork() *ethpb.Fork {
 
 // IsCanonical returns true if the input block root is part of the canonical chain.
 func (s *Service) IsCanonical(ctx context.Context, blockRoot [32]byte) (bool, error) {
-	// If the block has been finalized, the block will always be part of the canonical chain.
-	if s.cfg.BeaconDB.IsFinalizedBlock(ctx, blockRoot) {
-		return true, nil
+	isCanonical, err := s.cfg.ForkChoiceStore.IsCanonical(blockRoot)
+	if err == protoarray.ErrUnknownNodeRoot || err == doublylinkedtree.ErrNilNode {
```

Comment:
Using errors for existence will be easy to break. The desirable thing to do is change it to
```go
isCanonical, hasNode, err := IsCanonical(blockRoot)
...
func IsCanonical([]byte blockRoot) (bool, bool, error)
```

Reply:
I would rather have the caller call hasnode before than having this signature

Reply:
That's also good with me!